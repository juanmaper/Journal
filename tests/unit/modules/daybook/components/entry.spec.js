import { shallowMount } from "@vue/test-utils";
import Entry from '@/modules/daybook/components/Entry'
import { journalState } from '@/../tests/unit/mocks-data/test-journal-state';

describe('Entry component tests', () => {
  
  const mockRouter = {
    push: jest.fn()
  }

  const wrapper = shallowMount( Entry, {
    props: {
      entry: journalState.entries[0]
    },

    global: {
      mocks: {
        $router: mockRouter
      }
    }
  })

  test('should match the snapshot', () => {
    
    expect( wrapper.html() ).toMatchSnapshot()

  })

  test('should redirect when clicking the entry-container', () => {
    
    const entryContainer = wrapper.find('.entry-container')
    entryContainer.trigger('click')

    expect( mockRouter.push ).toHaveBeenCalledWith({
      name: 'entry',
      params: {
        id: journalState.entries[0].id
      }
    })
  })

  test('should have the right computed properties', () => {
    
    expect( wrapper.vm.day ).toBe(14)
    expect( wrapper.vm.month ).toBe('Junio')
    expect( wrapper.vm.yearDay ).toBe('2022, Martes')
    
  })

})