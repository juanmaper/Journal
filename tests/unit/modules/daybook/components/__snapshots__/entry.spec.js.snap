// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component tests should match the snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">14</span><span class="mx-1 fs-5">Junio</span><span class="mx-2 fw-light">2022, Martes</span></div>
  <div class="entry-description">Hello world from mock data</div>
</div>
`;
